lib/topology/platforms/shell.py

```diff
@@ -285,14 +296,15 @@ def __init__(
         self._timeout = timeout or -1
         self._encoding = encoding
         self._try_filter_echo = try_filter_echo
+        self._auto_connect = auto_connect
```

Comment:
*Because None as the default shells is just a way to say it that other connection. No special treatment.*

``None` ` as the default shell? Don't you mean ``None`` as the default value for ``connection``?